src/main/java/org/jabref/gui/BasePanel.java

```diff
@@ -772,10 +772,13 @@ public BibEntry newEntry(EntryType type) {
         EntryType actualType = type;
         if (actualType == null) {
             // Find out what type is wanted.
-            final EntryTypeDialog etd = new EntryTypeDialog(frame);
+            //final EntryTypeDialog etd = new EntryTypeDialog(frame);
```

Comment:
Is this method even used with null argument? At least the code below suggests that the JabRefFrame uses the NewEntryAction and does not calls this method.

Reply:
I just found out that this method and the insertEntry method are doing roughly the same, I think it makes sense to combine their functionaltiy